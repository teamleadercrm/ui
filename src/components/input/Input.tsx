import React, { ChangeEvent, forwardRef, useState } from 'react';
import { GenericComponent } from '../../@types/types';
import SingleLineInputBase, { SingleLineInputBaseProps } from './SingleLineInputBase';

interface InputProps extends Pick<SingleLineInputBaseProps, 'value'> {
  /** Callback function that is fired when the component's value changes. */
  onChange?: (event: ChangeEvent<HTMLElement>, value: string) => void;
}

const Input: GenericComponent<InputProps> = forwardRef<HTMLElement, InputProps>(
  ({ value, onChange, ...others }: InputProps, forwardedRef) => {
    const [stateValue, setStateValue] = useState(value);

    const handleChange = (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
      setStateValue(event.currentTarget.value);
      onChange && onChange(event, event.currentTarget.value);
    };

    return <SingleLineInputBase ref={forwardedRef} value={stateValue} onChange={handleChange} {...others} />;
  },
);

Input.displayName = 'Input';

export default Input;
