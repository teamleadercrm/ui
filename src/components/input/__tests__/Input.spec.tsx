import React from 'react';
import { render } from '@testing-library/react';
import Input from '../Input';

describe('Component - Input', () => {
  it('renders', async () => {
    const { asFragment } = render(<Input name="name" value="Michael Scott" placeholder="Dwight Schrute" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
