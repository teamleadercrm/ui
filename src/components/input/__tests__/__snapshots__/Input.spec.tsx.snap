// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Input renders 1`] = `
<DocumentFragment>
  <div
    class="box wrapper"
  >
    <div
      class="input-wrapper"
    >
      <div
        class="input-inner-wrapper"
      >
        <input
          class="input is-medium"
          name="name"
          placeholder="Dwight Schrute"
          type="text"
          value="Michael Scott"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
